VOM_Fortran/VOM-code/modules.f90

```diff
@@ -372,42 +375,58 @@ module vegmod
       REAL*8  :: lambdat_d              ! Target dE/dA for calculating gstomt (mol/mol)
       REAL*8  :: lambdag_d              ! Target dE/dA for calculating gstomg (mol/mol)
       REAL*8  :: gstomt                 ! Tree stomatal conductance (mol/m2/s)
-      REAL*8  :: gstomg(3,3,3)          ! Grass stomatal conductance (mol/m2/s)
-
-      REAL*8  :: rlt_h(3,3)             ! Tree leaf respiration for different values of Jmax (rlt_h(2) is actual value) (mol/h)
+      REAL*8  :: gstomts                ! Shaded tree stomatal conductance (mol/m2/s)      
+      REAL*8  :: gstomg(3,3,3)            ! Grass stomatal conductance (mol/m2/s)
+      REAL*8  :: gstomgs(3,3,3)           ! Shaded grass stomatal conductance (mol/m2/s)
+      
+      REAL*8  :: rlt_h(3)             ! Tree leaf respiration for different values of Jmax (rlt_h(2) is actual value) (mol/h)
+      REAL*8  :: rlts_h(3)            ! Shaded tree leaf respiration for different values of Jmax (rlt_h(2) is actual value) (mol/h)      
       REAL*8  :: rlt_d                  ! Daily tree leaf respiration (mol/d)
       REAL*8  :: rlt_y                  ! Annual tree leaf respiration (mol/y)
-      REAL*8  :: rlg_h(3,3,3)           ! Grass leaf respiration (mol/h)
+      REAL*8  :: rlg_h(3)           ! Grass leaf respiration (mol/h)
```

Comment:
Why does rlg_h now only have dimension 3, not 3,3,3? I thought that with sunlit and shaded distinctions, there should be more, not fewer dimensions.

Reply:
Previously, it was using three daily optimized values: jmax25g, grass cover, and LAI. The mutliplication with grass cover is now at the end. The LAI previously led to a fraction of absorbed radiation, which was also part of the Rl calculations, but we discussed this some time ago and concluded it shouldn't be there. It used to be this:

```
rlg_h(1,:,ii) = ((ca_h(th_) - gammastar) * pcg_d(1) * Ma_lg(ii) * jmaxg_h(:)    &
        &           * i_rlratio) / (4.d0 * (ca_h(th_) + 2.d0 * gammastar) &
        &           * (1.d0 + i_rlratio))  ! (3.24), (Out[312])
```
where Ma_lg was the fraction of absorbed radiation as a function of LAI. Now it is just this:

```
!    * respiration grasses
         rlg_h(:) = ((ca_h(th_) - gammastar) * jmaxg_h(:)    &
        &           * i_rlratio) / (4.d0 * (ca_h(th_) + 2.d0 * gammastar) &
        &           * (1.d0 + i_rlratio))  ! (3.24), (Out[312])
```

So now there is just the dimension 3, just due to the optimization of jmax25. There is now also a shaded leaf respiration, which I named rlgs_h (additional s).